### ISSUE TITLE
Messages are not delivered to gmail accounts

### ISSUE THREAD

**Issue by mtarunsai1999** [LukeSmithxyz/emailwiz#21]
I have set this up on debian server. I use thunderbird client. It receives mail fine. But the mails it shows as sent are not delivered to recipients (gmail account holders) not even as spam. I checked the logs and it says connection to google server timed out. Any help on what to do? I do have DKIM TXT record.

**Comment by LukeSmithxyz** [comment]
Some possibilities:

- If you ran the script more than once, you might've ended up with two different DKIM entries and you might need to change your DNS settings to the other instead.
- Is it specifically saying that port 25 is timing out? I've had this error before and people have given it, although postfix should be sending mail on 587 so it's encrypting. Gmail might reject port 25.

**Comment by Oborichkin** [comment]
Same issue here.
I'm using Ubuntu 20.04 and to properly start OpenDKIM I had to do:
```bash
/lib/opendkim/opendkim.service.generate
systemctl daemon-reload
systemctl restart opendkim
```
Postfix also was complaining about missing `/etc/securetty`. Solved this by manualy creating it and copying it's contents from 19.10.
Now I'm getting conection timeout for all servers, not just Gmail. Everytime its port 25.

**Comment by mtarunsai1999** [comment]
My dkim entry is correct. 

This is the error in the logifles 

`Jun 19 10:58:04 mail postfix/smtp[21698]: 9593720EC0: to=<-------@gmail.com>, relay=none, delay=239755, delays=239604/0.03/150/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[2607:f8b0:4003:c0f::1b]:25: Connection timed out)`

**Comment by Mattrez** [comment]
I'd like to chip in saying that if you are using AWS to host your VPS they block port 25 by default. So if you want to send mail while using AWS as a host provider they have a request to fill stating your reason.

**Comment by LukeSmithxyz** [comment]
I just retried this on a new Vultr server (for a video) and it worked perfectly.

There are two other possibilities that occured to me as to what could be wrong in the people's setups that have this problem:

- I noticed that my server attempts and suceeds in sending to Gmail and some sites via IPv6, you might want to have an IPv6 address and an AAAA record for it, just in case.
- Due to unclarity on my part, there are two little TXT entries that should be entered after the script is cover: one is the big `v=DKIM1; ...` one, but there's also one that for me looks like this: `v=spf1 mx a:mail.unaboomer.com -all` with the host of `@`.

**Comment by benf3632** [comment]
I tried putting the two txt records but still no luck on sending mails to other mail servicer.

**Comment by mtarunsai1999** [comment]
I had AAAA and spf TXT records. I'll try reinstalling.

**Comment by benf3632** [comment]
From what I have understood that vultr blocks outgoing port 25. I contacted the support in an effort to unblock the port.

**Comment by mtarunsai1999** [comment]
Vultr did block smtp for my VPS. I contacted support staff. Now everything works.

**Comment by mtarunsai1999** [comment]
Blocking smtp seems to be the default policy of VPS services. This probably needs a mention in the README.

**Comment by LukeSmithxyz** [comment]
That's weird... Vultr has never blocked my 25, but I'll put it in the README just in case.

**Comment by GNUke** [comment]
Vultr also blocked port 25 on my instance, support seems to be pretty reluctant to unblock it. 

They responded to me asking for the volume of emails i estimate will be sent per day/month so you should include that in your initial ticket if you want to get it unblocked.

**Comment by FlyingKawasaki74** [comment]
To extend the info by @GNUke , Vultr support sent me a detailed list of what they require before unblocking port 25:

_Please reply to this ticket with the following information:
1. The business name and organization URL(s) under which you offer services.
2. Describe, in as much detail as possible, the nature of the emails you intend to send.
3. The volume of email that you plan to deliver on a daily/monthly basis._

They might have gotten a bit worried about the influx of mail servers after Luke's video hahaha
